src/tools.cpp

```diff
@@ -454,8 +454,12 @@ void asciitolower(std::string& s)
 UriKind html_link::detectUriKind(const std::string& input_string)
 {
     const auto k = input_string.find_first_of(":/?#");
-    if ( k == std::string::npos || input_string[k] != ':' )
-        return UriKind::OTHER;
+    if ( k == std::string::npos || input_string[k] != ':' ) {
+        if ( k == 0 && input_string.substr(0, 2) == "//" )
```

Comment:
> This will fail if `input_string` is "/".

No, it won't. [`std::string::substr(pos, count)`](https://en.cppreference.com/w/cpp/string/basic_string/substr) is immune to such usage - if the requested substring extends past the end of the string, i.e. the count is greater than `size() - pos` (e.g. if `count == npos`), the returned substring is [`pos`, `size()`).

Reply:
> (And it would be good to add a test for `uriKind("/")`)

Done

> It would be better to add a test on `k==0` in the following test returning (for now) `UriKind::GENERIC_UR`

You can't get along with that alone. The first check has to be modified too.

Reply:
> No, it won't. [std::string::substr(pos, count)](https://en.cppreference.com/w/cpp/string/basic_string/substr) is immune to such usage - if the requested substring extends past the end of the string, i.e. the count is greater than size() - pos (e.g. if count == npos), the returned substring is [pos, size()).

Indeed. Then we are good.